### PR TITLE
Adding oas-apigee-mock lab

### PR DESCRIPTION
**Pull request by markjkelly**
What's changed, or what was fixed?

- Added a codelab which walks through the process of generating, deploying and testing a mock 
proxy on Apigee using oas-apigee-mock, apigee-sackmesser and apickli, along with other utilities.

- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

File: labs/oas-apigee-mock/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Apigee OAS Mock Lab
+
```
Comment (danistrebel) [inline labs/oas-apigee-mock/README.md:2]: Maybe add a sentence on what this is all about. Happy if you want to just copy the one from the main readme. I think this captures it nicely.

File: labs/oas-apigee-mock/lab.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
+id: oas-apigee-mock
+
+# OAS Apigee Mock Lab
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:3]: I am thinking maybe change the theme to TDD or BDD proxy development instead?

The mocking part is one aspect but IMO you're telling a bigger story here.

File: labs/oas-apigee-mock/lab.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+[API Jam](https://github.com/apigee/apijam) will cover the same topics in
+less depth.
+
+Lets get started!
+
+## Prerequisites
+
+Duration: 0:30:00
+
+An Apigee X organisation configured for external exposure is required to
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:38]: X or hybrid? up to you if you want to go into this discussion or not

File: labs/oas-apigee-mock/lab.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+Clone the Apigee DevRel repo and install the dependencies for the
+oas-apigee-mock tool.
+
+``` bash
+git clone https://github.com/apigee/devrel.git
+cd devrel/tools/oas-apigee-mock/
+npm install
+```
+
+## Generate, deploy and test an Apigee Proxy
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:79]: Maybe split these into separate steps?

File: labs/oas-apigee-mock/lab.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+Duration: 0:30:00
+
+### Generate an Apigee Proxy from an Open API Specification
+
+Use the `orders-apikey-header.yaml` Open API specification included
+in the `devrel/tools/oas-apigee-mock/test/oas` folder to generate a proxy bundle.
+
+``` bash
+node bin/oas-apigee-mock generateApi web-orders-proxy-v1 \
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:89]: specify from where this command should be run

File: labs/oas-apigee-mock/lab.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+  -s test/oas/orders-apikey-header.yaml \
+  -o
+```
+
+### Update the apickli configuration to use your Host Alias
+
+Update the `before(function ()` block in
+`test/features/step_definitions/init.js` with your organisation's hostname.
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:97]: host names are at the env group level.

File: labs/oas-apigee-mock/lab.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+This will be your `RUNTIME_HOST_ALIAS` if you followed the
+[Apigee X Trial Provisioning](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+script.
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:100]: Not entirely clear what `This` is referencing here. Maybe you can put in the curl command to list the envgroups and jq to extract the hostnames instead?

File: labs/oas-apigee-mock/lab.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  );
+```
+
+### Update the test suite proxy reference
+
+Update the test suite with your proxy name, we used `web-orders-proxy-v1` in
+the generate proxy step.
+
+``` bash
+sed -i 's/oas-apigee-mock-orders-apikey-header/web-orders-proxy-v1/' test/features/orders-apikey-header.feature
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:115]: before you asked for a manual edit of the file. Do you think it makes sense to also ask for this here to be consistent and have the user look at the full config?

File: labs/oas-apigee-mock/lab.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+./node_modules/.bin/cucumber-js test/features/orders-apikey-header.feature --format json:test/test_report.json --publish-quiet
+```
+
+The test should fail as we have yet to deploy the proxy we generated
+previously.
+
+### Deploy the generated proxy to your Apigee Organisation
+
+``` bash
+sackmesser deploy -d "$PWD/api_bundles/web-orders-proxy-v1" --googleapi -t "$TOKEN" -o "$APIGEE_ORG" -e "$APIGEE_ENV"
```
Comment (danistrebel) [inline labs/oas-apigee-mock/lab.md:132]: Maybe replace $PWD with a variable that we set earlier like `$LAB_HOME` such that the user can move around if they want to

**Review by danistrebel — CHANGES_REQUESTED** [review]
Thanks Mark, great BDD walkthrough. I added a few suggestions. Please let me know what you think about them

File: README.md [hunk 10]
```diff
@@ -85,6 +85,9 @@ technical or non-technical topic.
 
 - [Best Practices Hackathon](labs/best-practices-hackathon) [(web)](https://apigee.github.io/devrel/labs/best-practices-hackathon)
   A 300 level lab to learn Apigee Best Practices
+- [OAS Apigee Mock](labs/oas-apige-mock) [(web)](https://apigee.github.io/devrel/labs/oas-apigee-mock)
```
Comment (seymen) [inline README.md:88]: `oas-apige` -> `oas-apigee`

**Comment by markjkelly** [comment]
Thanks for the review and feedback @danistrebel and @seymen, I've made some changes, let me know what you think.

File: labs/bdd-proxy-development/lab.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+./node_modules/.bin/cucumber-js test/features/orders-apikey-header.feature --format json:test/test_report.json --publish-quiet
+```
+
+The test should fail as we have yet to deploy the proxy we generated
+previously.
+
+### Deploy the generated proxy to your Apigee Organisation
+
+``` bash
+sackmesser deploy -d "$LAB_HOME/api_bundles/web-orders-proxy-v1" --googleapi -t "$TOKEN" -o "$APIGEE_ORG" -e "$APIGEE_ENV"
```
Comment (danistrebel) [inline labs/bdd-proxy-development/lab.md:157]: @markjkelly to reduce the number of different tools you might want to consider doing this via the apigeecli tool as well. I'll leave it up to you.

**Review by danistrebel — APPROVED** [review]
LGTM

File: labs/bdd-proxy-development/lab.md (hunk 11 above)
Comment (OmidTahouri) [reply labs/bdd-proxy-development/lab.md:157]: +1 but not a blocker

**Comment by OmidTahouri** [comment]
LGTM

**Pull request merged**